### DISCUSSION TITLE
Error: Command "yarn run build" exited with 1 (WHY LAAAHD WHYY!!)

### DISCUSSION THREAD

**Discussion by NolvusMadeIt** [vercel/community#1538] [Help]
### Summary

Im not undersztanding this. Its working locally but when installing it on vercel im getting: `Error: Command "yarn run build" exited with 1`

Help is so much appriciated, Ive been at this for owls!

### Additional information

Running build in Washington, D.C., USA (East) – iad1
02:53:54.500 | Cloning github.com/HelloMikko/MyPortfolio (Branch: main, Commit: fef7c9e)
02:53:54.506 | Skipping build cache since Vercel CLI used --force option
02:53:55.055 | Cloning completed: 555.232ms
02:53:55.177 | Running "vercel build"
02:53:55.631 | Vercel CLI 28.15.3
02:53:55.920 | Installing dependencies...
02:53:56.220 | yarn install v1.22.17
02:53:56.300 | warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
02:53:56.303 | [1/4] Resolving packages...
02:53:56.791 | [2/4] Fetching packages...
02:54:15.283 | [3/4] Linking dependencies...
02:54:15.286 | warning "@lyket/react > @emotion/react > @emotion/babel-plugin@11.10.5" has unmet peer dependency "@babel/core@^7.0.0".
02:54:15.287 | warning "@lyket/react > @emotion/react > @emotion/babel-plugin > @babel/plugin-syntax-jsx@7.18.6" has unmet peer dependency "@babel/core@^7.0.0-0".
02:54:15.288 | warning " > next-auth@4.19.2" has incorrect peer dependency "next@^12.2.5 \|\| ^13".
02:54:15.289 | warning " > react-rough-notation@1.0.3" has incorrect peer dependency "react@^18.2.0".
02:54:15.289 | warning " > react-rough-notation@1.0.3" has incorrect peer dependency "react-dom@^18.2.0".
02:54:15.294 | warning "eslint-config-next > @typescript-eslint/parser > @typescript-eslint/typescript-estree > tsutils@3.21.0" has unmet peer dependency "typescript@>=2.8.0 \|\| >= 3.2.0-dev \|\| >= 3.3.0-dev \|\| >= 3.4.0-dev \|\| >= 3.5.0-dev \|\| >= 3.6.0-dev \|\| >= 3.6.0-beta \|\| >= 3.7.0-dev \|\| >= 3.7.0-beta".
02:54:15.295 | warning " > file-loader@6.2.0" has unmet peer dependency "webpack@^4.0.0 \|\| ^5.0.0".
02:54:30.001 | [4/4] Building fresh packages...
02:54:35.416 | success Saved lockfile.
02:54:35.420 | $ prisma generate \|\| true
02:54:36.578 | Prisma schema loaded from prisma/schema.prisma
02:54:37.782 |  
02:54:37.782 | ✔ Generated Prisma Client (3.15.2 \| library) to ./node_modules/@prisma/client in 87ms
02:54:37.783 | You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
02:54:37.783 | ```
02:54:37.783 | import { PrismaClient } from '@prisma/client'
02:54:37.783 | const prisma = new PrismaClient()
02:54:37.783 | ```
02:54:37.857 | $ husky install
02:54:37.948 | husky - Git hooks installed
02:54:37.952 | Done in 41.74s.
02:54:37.984 | Detected Next.js version: 12.1.4
02:54:38.000 | Running "yarn run build"
02:54:38.177 | yarn run v1.22.17
02:54:38.220 | $ next build && node ./scripts/generate-sitemap
02:54:38.888 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
02:54:38.888 | This information is used to shape Next.js' roadmap and prioritize features.
02:54:38.888 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
02:54:38.888 | https://nextjs.org/telemetry
02:54:38.889 |  
02:54:39.052 | info  - Checking validity of types...
02:54:45.474 | info  - Creating an optimized production build...
02:54:47.130 |  
02:54:47.130 | warn - You have enabled experimental features: optimizeUniversalDefaults
02:54:47.131 | warn - Experimental features in Tailwind CSS are not covered by semver, may introduce breaking changes, and can change at any time.
02:55:19.349 | info  - Compiled successfully
02:55:19.349 | info  - Collecting page data...
02:55:22.245 | /vercel/path0/node_modules/@prisma/client/runtime/index.js:44703
02:55:22.245 | throw new PrismaClientInitializationError(error2.message, this.config.clientVersion, error2.error_code);
02:55:22.245 | ^
02:55:22.246 |  
02:55:22.247 | PrismaClientInitializationError: error: Environment variable not found: DATABASE_URL.
02:55:22.247 | -->  schema.prisma:8
02:55:22.247 | \|
02:55:22.247 | 7 \|   provider             = "mysql"
02:55:22.247 | 8 \|   url                  = env("DATABASE_URL")
02:55:22.248 | \|
02:55:22.248 |  
02:55:22.248 | Validation Error Count: 1
02:55:22.248 | at LibraryEngine.loadEngine (/vercel/path0/node_modules/@prisma/client/runtime/index.js:44703:19)
02:55:22.248 | at async LibraryEngine.instantiateLibrary (/vercel/path0/node_modules/@prisma/client/runtime/index.js:44632:5) {
02:55:22.248 | clientVersion: '3.15.2',
02:55:22.248 | errorCode: 'P1012'
02:55:22.248 | }
02:55:22.248 |  
02:55:22.248 | Node.js v18.12.1
02:55:23.455 | info  - Generating static pages (0/32)
02:55:23.609 | info  - Generating static pages (8/32)
02:55:24.027 | (node:927) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
02:55:24.029 | (Use `node --trace-warnings ...` to show where the warning was created)
02:55:24.098 |  
02:55:24.099 | Error occurred prerendering page "/tweets". Read more: https://nextjs.org/docs/messages/prerender-error
02:55:24.099 | TypeError: Cannot read properties of undefined (reading 'reduce')
02:55:24.100 | at getTweets (/vercel/path0/.next/server/pages/tweets.js:295:24)
02:55:24.100 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
02:55:24.100 | at async getStaticProps (/vercel/path0/.next/server/pages/tweets.js:332:20)
02:55:24.101 | at async renderToHTML (/vercel/path0/node_modules/next/dist/server/render.js:470:20)
02:55:24.101 | at async /vercel/path0/node_modules/next/dist/export/worker.js:253:36
02:55:24.101 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
02:55:24.321 | info  - Generating static pages (16/32)
02:55:24.435 |  
02:55:24.436 | Error occurred prerendering page "/guestbook". Read more: https://nextjs.org/docs/messages/prerender-error
02:55:24.436 | Error: error: Environment variable not found: DATABASE_URL.
02:55:24.437 | -->  schema.prisma:8
02:55:24.437 | \|
02:55:24.437 | 7 \|   provider             = "mysql"
02:55:24.437 | 8 \|   url                  = env("DATABASE_URL")
02:55:24.437 | \|
02:55:24.438 |  
02:55:24.438 | Validation Error Count: 1
02:55:24.439 | at RequestHandler.request (/vercel/path0/node_modules/@prisma/client/runtime/index.js:49028:15)
02:55:24.440 | at async PrismaClient._request (/vercel/path0/node_modules/@prisma/client/runtime/index.js:49919:18)
02:55:24.441 | at async getStaticProps (/vercel/path0/.next/server/pages/guestbook.js:454:21)
02:55:24.441 | at async renderToHTML (/vercel/path0/node_modules/next/dist/server/render.js:470:20)
02:55:24.441 | at async /vercel/path0/node_modules/next/dist/export/worker.js:253:36
02:55:24.442 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
02:55:24.449 | info  - Generating static pages (24/32)
02:55:24.535 |  
02:55:24.535 | Error occurred prerendering page "/blog/Apple-Competition". Read more: https://nextjs.org/docs/messages/prerender-error
02:55:24.536 | Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
02:55:24.536 | at b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:45:32)
02:55:24.536 | at b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
02:55:24.536 | at exports.renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
02:55:24.537 | at Object.renderPage (/vercel/path0/node_modules/next/dist/server/render.js:768:45)
02:55:24.537 | at Object.defaultGetInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:375:51)
02:55:24.537 | at MyDocument.getInitialProps (/vercel/path0/.next/server/chunks/6859.js:566:20)
02:55:24.537 | at Object.loadGetInitialProps (/vercel/path0/node_modules/next/dist/shared/lib/utils.js:65:29)
02:55:24.538 | at documentInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:781:48)
02:55:24.538 | at renderDocument (/vercel/path0/node_modules/next/dist/server/render.js:806:55)
02:55:24.538 | at renderToHTML (/vercel/path0/node_modules/next/dist/server/render.js:906:34)
02:55:24.624 |  
02:55:24.624 | Error occurred prerendering page "/blog/WWDC22". Read more: https://nextjs.org/docs/messages/prerender-error
02:55:24.625 | Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
02:55:24.625 | at b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:45:32)
02:55:24.625 | at b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
02:55:24.625 | at exports.renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
02:55:24.625 | at Object.renderPage (/vercel/path0/node_modules/next/dist/server/render.js:768:45)
02:55:24.625 | at Object.defaultGetInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:375:51)
02:55:24.625 | at MyDocument.getInitialProps (/vercel/path0/.next/server/chunks/6859.js:566:20)
02:55:24.625 | at Object.loadGetInitialProps (/vercel/path0/node_modules/next/dist/shared/lib/utils.js:65:29)
02:55:24.625 | at documentInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:781:48)
02:55:24.626 | at renderDocument (/vercel/path0/node_modules/next/dist/server/render.js:806:55)
02:55:24.626 | at renderToHTML (/vercel/path0/node_modules/next/dist/server/render.js:906:34)
02:55:24.712 |  
02:55:24.712 | Error occurred prerendering page "/blog/spotify-refresh-token". Read more: https://nextjs.org/docs/messages/prerender-error
02:55:24.712 | Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
02:55:24.712 | at b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:45:32)
02:55:24.712 | at b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
02:55:24.712 | at exports.renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
02:55:24.712 | at Object.renderPage (/vercel/path0/node_modules/next/dist/server/render.js:768:45)
02:55:24.712 | at Object.defaultGetInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:375:51)
02:55:24.712 | at MyDocument.getInitialProps (/vercel/path0/.next/server/chunks/6859.js:566:20)
02:55:24.712 | at Object.loadGetInitialProps (/vercel/path0/node_modules/next/dist/shared/lib/utils.js:65:29)
02:55:24.712 | at documentInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:781:48)
02:55:24.712 | at renderDocument (/vercel/path0/node_modules/next/dist/server/render.js:806:55)
02:55:24.713 | at renderToHTML (/vercel/path0/node_modules/next/dist/server/render.js:906:34)
02:55:25.887 |  
02:55:25.887 | Error occurred prerendering page "/blog/Apple-Privacy". Read more: https://nextjs.org/docs/messages/prerender-error
02:55:25.887 | Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
02:55:25.888 | at b.render (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:45:32)
02:55:25.888 | at b.read (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
02:55:25.888 | at exports.renderToString (/vercel/path0/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
02:55:25.888 | at Object.renderPage (/vercel/path0/node_modules/next/dist/server/render.js:768:45)
02:55:25.888 | at Object.defaultGetInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:375:51)
02:55:25.888 | at MyDocument.getInitialProps (/vercel/path0/.next/server/chunks/6859.js:566:20)
02:55:25.889 | at Object.loadGetInitialProps (/vercel/path0/node_modules/next/dist/shared/lib/utils.js:65:29)
02:55:25.892 | at documentInitialProps (/vercel/path0/node_modules/next/dist/server/render.js:781:48)
02:55:25.892 | at renderDocument (/vercel/path0/node_modules/next/dist/server/render.js:806:55)
02:55:25.892 | at renderToHTML (/vercel/path0/node_modules/next/dist/server/render.js:906:34)
02:55:25.893 |  
02:55:25.893 | > Build error occurred
02:55:25.893 | Error: Export encountered errors on following paths:
02:55:25.893 | /blog/[...slug]: /blog/Apple-Competition
02:55:25.893 | /blog/[...slug]: /blog/Apple-Privacy
02:55:25.894 | /blog/[...slug]: /blog/WWDC22
02:55:25.894 | /blog/[...slug]: /blog/spotify-refresh-token
02:55:25.894 | /guestbook
02:55:25.894 | /tweets
02:55:25.894 | at /vercel/path0/node_modules/next/dist/export/index.js:397:19
02:55:25.894 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
02:55:25.895 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
02:55:25.895 | at async /vercel/path0/node_modules/next/dist/build/index.js:999:17
02:55:25.895 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
02:55:25.895 | at async /vercel/path0/node_modules/next/dist/build/index.js:875:13
02:55:25.895 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
02:55:25.896 | at async Object.build [as default] (/vercel/path0/node_modules/next/dist/build/index.js:55:25)
02:55:25.896 | info  - Generating static pages (32/32)
02:55:25.979 | error Command failed with exit code 1.
02:55:25.979 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
02:55:25.997 | Error: Command "yarn run build" exited with 1
02:55:26.066

**Comment by amyegan** [comment]
It looks like you're missing the `DATABASE_URL` environment variable.
> PrismaClientInitializationError: error: Environment variable not found: DATABASE_URL.

And that seems to be causing build errors on a few pages:
> 02:55:25.893 | > Build error occurred
02:55:25.893 | Error: Export encountered errors on following paths:
02:55:25.893 | /blog/[...slug]: /blog/Apple-Competition
02:55:25.893 | /blog/[...slug]: /blog/Apple-Privacy
02:55:25.894 | /blog/[...slug]: /blog/WWDC22
02:55:25.894 | /blog/[...slug]: /blog/spotify-refresh-token
02:55:25.894 | /guestbook
02:55:25.894 | /tweets

`Error: Command "yarn run build" exited with 1` will most likely go away if you address the other issues. You probably only need to set `DATABASE_URL` via the [environment variables](https://vercel.com/docs/concepts/projects/environment-variables#) UI in your project settings to get rid of the error.

**Reply by NolvusMadeIt** [reply]
Hey thank you @amyegan however, I am not using any db's at all, could I just use a dummy url or something? Also, I use npm instead of yarn. However, I did just run yarn and it seemed to start building. However Im sure it would not solve the problem.Lastly, I am unsure on what to put in these fields as I have never used nextjs on vercel before: <img src="https://i.imgur.com/YeqCnP7.png"></img>

**Comment by denyskotsur** [comment]
@codemikko Any updates on this?

**Comment by heozeop** [comment]
is upgrade cli version works?
